### ISSUE TITLE
CUDA 10 / CUDNN 7.5 performance loss (Titan V)

### ISSUE THREAD

**Issue by MartinSmeyer** [tensorflow/tensorflow#27623]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): openSUSE Leap 42.3
- GPU model and memory: Titan V
- GCC/Compiler version (if compiling from source): 4.8
- Nvidia Driver Version: 418.56

**Problem:** I experience consistent performance losses in TF using CUDA 10.0/CUDNN 7.5. 
Here is a minimal example:

```
import tensorflow as tf
import time

matrix_dim = (5120,5120)
noof_matmul = 512
num_gpus = 1
float_type = tf.float32
c = []

print(tf.__file__)
for d in range(num_gpus):
    with tf.device('/gpu:%s' % d):
        a=tf.random_normal(matrix_dim,dtype=float_type)
        for i in range(noof_matmul):
            a=tf.matmul(a,a)
        c.append(a.op)

with tf.Session() as sess:
    sess.run(c)
    start=time.time()
    for i in range(10):
        sess.run(c)

t = (time.time()-start)/10.
tflops = noof_matmul * (2 * matrix_dim[0]**3 - matrix_dim[0]**2) / t / 10**12
print(t, 'sec, ', tflops, 'TFlops')
```

**Results Titan V:**
Tensorflow 1.13.1 built from source, bazel 0.24.0, CUDA 10.0, CUDNN 7.5.0:
```diff
-FP16: 1.6899 sec, 81.3209 TFlops
-FP32: 10.6873 sec, 12.8588 TFlops
```
Tensorflow 1.9 built from source (same configuration besides versions), bazel 0.11.1, CUDA 9.1, CUDNN 7.1.4:
```diff
+FP16: 1.5391 sec, 89.2890 TFlops
+FP32: 10.1137 sec, 13.5881 TFlops
```
Pip wheel tensorflow_gpu-1.13.1-cp27-cp27mu-manylinux1_x86_64.whl, CUDA 10.0, CUDNN 7.5.0:
```diff
-FP16: 1.6902 sec, 81.3057 TFlops
-FP32: 10.5577 sec, 13.01656 TFlops
```
Conda (Python 3.6) linux-64/tensorflow-gpu-1.13.1-h0d30ee6_0.tar.bz2, CUDA 9.2, CUDNN 7.3.1:
```diff
+FP16: 1.5389 sec, 89.2995 TFlops
+FP32: 10.1150 sec, 13.5863 TFlops
```

Have you experienced similar performance losses? Do you know any solutions? If necessary, I can run tests on other GPU models as well.

**Comment by zkzhu0110** [comment]
Could I ask a question? When I turn "tf.float32" into "tf.float64" , this program cannot be stopped. How can I calculate FP64 for our GPUs? Thank you

**Comment by byronyi** [comment]
Leaving this kind of performance numbers here is not very helpful. Neither this particular microbenchmark is important nor it is likely to be fixed in old releases, e.g. a TF 1.13.2 wheel is not likely to be released even if there is a fix for this.

I will suggest you use the nightly package and benchmark again.

**Comment by sanjoy** [comment]
This is fixed in tf-nightly (could have been a TF fix or because we're now using CUDA 10.1).

On TF 2.0.0:

fp16: 81.29 tflops
fp32: 13.12 tflops

On TF 2.1.0-dev20191226

fp16: 93.51 tflops
fp32: 13.58 tflops

All numbers on Titan-V.

**Comment by sanjoy** [comment]
> Could I ask a question? When I turn "tf.float32" into "tf.float64" , this program cannot be stopped. How can I calculate FP64 for our GPUs? Thank you

@zkzhu0110 please open a separate github issue for this if your question is still relevant.
